admin/class-primary-term-admin.php

```diff
@@ -0,0 +1,195 @@
+	}
+
+	/**
+	 * Returns an array suitable for use in the javascript
+	 *
+	 * @param stdClass $term The term to map.
+	 *
+	 * @return array
+	 */
+	private function map_terms_for_js( $term ) {
+		return array(
+			'id'   => $term->term_id,
+			'name' => $term->name,
+		);
+	}
+
+	/**
+	 * Save the primary term for a specific taxonomy
+	 *
+	 * @param int      $post_ID  Post ID to save primary term for.
+	 * @param stdClass $taxonomy Taxonomy to save primary term for.
+	 */
+	private function save_primary_term( $post_ID, $taxonomy ) {
+		$primary_term = filter_input( INPUT_POST, WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_term', FILTER_SANITIZE_NUMBER_INT );
+
+		// We accept an empty string here because we need to save that if no terms are selected.
+		if ( null !== $primary_term && check_admin_referer( 'save-primary-term', WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_nonce' ) ) {
+			$primary_term_object = new WPSEO_Primary_Term( $taxonomy->name, $post_ID );
+			$primary_term_object->set_primary_term( $primary_term );
+		}
+	}
+
+	/**
+	 * Returns all the taxonomies for which the primary term selection is enabled
+	 *
+	 * @param int $post_ID Default current post ID.
+	 * @return array
+	 */
+	private function get_primary_term_taxonomies( $post_ID = null ) {
+
+		if ( null === $post_ID ) {
+			$post_ID = get_the_ID();
+		}
+
+		if ( false !== ( $taxonomies = wp_cache_get( 'primary_term_taxonomies_' . $post_ID, 'wpseo' ) ) ) {
+			return $taxonomies;
+		}
+
+		$post_type      = get_post_type( $post_ID );
+		$all_taxonomies = get_object_taxonomies( $post_type, 'objects' );
+		$all_taxonomies = array_filter( $all_taxonomies, array( $this, 'filter_hierarchical_taxonomies' ) );
+		$taxonomies     = array_filter( $all_taxonomies, array( $this, 'filter_category_taxonomy' ) );
+
+		/**
+		 * Filters which taxonomies for which the user can choose the primary term. Only category is enabled by default.
+		 *
+		 * @api array    $taxonomies An array of taxonomy objects that are primary_term enabled.
+		 * @param string $post_type The post type for which to filter the taxonomies.
+		 * @param array  $all_taxonomies All taxonomies for this post types, even ones that don't have primary term
+		 *                              enabled.
+		 */
+		$taxonomies = (array) apply_filters( 'wpseo_primary_term_taxonomies', $taxonomies, $post_type, $all_taxonomies );
+
+		wp_cache_set( 'primary_term_taxonomies_' . $post_ID, $taxonomies, 'wpseo' );
+
+		return $taxonomies;
+	}
+
+	/**
+	 * Returns whether or not a taxonomy is hierarchical
+	 *
+	 * @param stdClass $taxonomy Taxonomy object.
+	 *
+	 * @return bool
+	 */
+	private function filter_hierarchical_taxonomies( $taxonomy ) {
+		return true === $taxonomy->hierarchical;
+	}
+
+	/**
+	 * Returns whether or not the taxonomy is the category taxonomy
+	 *
+	 * @param stdClass $taxonomy Taxonomy object.
+	 *
+	 * @return bool
+	 */
+	private function filter_category_taxonomy( $taxonomy ) {
```

Comment:
Probably PHP will give a warning because it is trying to access a method outside the scope of the class. Probably this method has to be `public`.